### PR TITLE
Drop autoescaping

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes/no
| New feature?  | yes/no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

See also https://github.com/trasher/glpi/pull/449 I've kept separate to limit changes here... It will maybe ease reviews.

### CODE REVIEW

**Comment by trasher** [comment]
There are still some tests that are failing; but a first review pass would probably help. ping @orthagh @cedric-anne ; we can speak about all that on our work channels ;)

**Review by cedric-anne — COMMENTED** [review]
As seen IRL, here are the following steps to achieve.

1. Deprecate usage of `DBmysql::query()` method. It could be replaced by a new method `DBmysql::doQuery()`. This new method could do the exact same thing, or could expect an instance of a new `\Glpi\Db\RawQuery` object as unique parameter. Anyway, the idea it to forece developpers review all usages of `DBmysql::query()` in order to either replace them by `DBmysql::request()` to benefits from SQL injection protection, either unsing the new `DBmysql::doQuery()` for call that are considered safe and are not currently supported by `DBmysql::request()` (e.g. `SELECT VERSION()`).

2. Add an automatic call to `\Glpi\Toolbox\Sanitizer::decodeHtmlSpecialChars()` on all string values fetched from database. This logic should probably be put in `DBmysqlIterator::current()` method.

Additionaly, I think we should remove all calls to `Sanitizer::sanitize()` that are made prior to `CommonDBTM::add()` and `CommonDBTM::update()` calls and remove all calls to `|verbatim_value` in Twig templates.

Once this would be done, we should be able to merge this PR, but we will have to then review all `echo` usages in GLPI to adapt code against XSS (either by replacing them by a Twig template, either by using `htmlspecialchars()` function) :fearful: 
We will also have to review our javascript code to ensure that presence of raw HTML special chars in dynamic values is correctly handled.

File: install/migrations/update_9.1.x_to_9.2.0.php [hunk 1]
```diff
@@ -215,17 +215,30 @@ function update91xto920()
...
+                            'glpi_documents_items' => 'documents_id',
+                            'glpi_documents' => 'id'
+                        ]
+                    ]
+                ]
+            ]
+        ),
+        "9.2 update set users_id on glpi_documents_items",
+        [],
+        true
```
[the remark this answers is not in the document]
Comment (trasher) [reply install/migrations/update_9.1.x_to_9.2.0.php]: This is not strictly required, it was an old todo I've never have time to work on. This is also a quite simple change, that why I did it.

File: src/CronTask.php [hunk 2]
```diff
@@ -397,15 +397,15 @@ public function getNeedToRun($mode = 0, $name = '')
          // Core crontasks
             [
                 ['NOT' => ['itemtype' => ['LIKE', 'Plugin%']]],
-                ['NOT' => ['itemtype' => ['LIKE', addslashes('GlpiPlugin\\\\') . '%']]]
+                ['NOT' => ['itemtype' => ['LIKE', 'GlpiPlugin\\\\' . '%']]]
```
[the remark this answers is not in the document]
Comment (trasher) [reply src/CronTask.php:400]: It may require a test, yes, but I do not exactly know what is expected

[the remark this answers is not in the document]
Comment (trasher) [reply src/CronTask.php:400]: I do not know for plugins, but for core crontasks using a namespace, they're stored in their more simple form (`Glpi\Marketplace\Controller` for example).
Using addslashes here gives `GlpiPlugin\\`, while removng it gives `GlpiPlugin\`. The second one seems correct.

**Comment by cedric-anne** [comment]
Replaced by #14302 that is using a branch from current repository in order to permit to open pull requests here.

**Pull request closed without merging**